### DISCUSSION TITLE
Negative weights

### DISCUSSION THREAD

**Discussion by SinaSabzevar** [krishnanlab/PecanPy#319] [Q&A]
Hi
I am using pecanpy for embeding causal graph that contains negative weights but pecanpy ignores the negative weights is there any solution for this issue?

**Comment by RemyLau** [comment]
Hi @SinaSabzevar, thanks using PecanPy! I would say it depends on how you want to treat the negative edges. For example, if you treat negative edges just the same type of interaction as edges with positive edge weights, then you could simply add a preprocessing step that converts all edge weights to the absolute value.

But if you want to treat positive and negative edges as two different edge types, you could also potentially create two seperate graphs, one only with positive edges, and the other with negative ones. By doing so, you will end up with two graphs, each only with positive edge weights. You could then embed the two graphs separately and combine the embeddings as you like, e.g., concatenating.

There are many other ways to work around, but it all comes down to a research question: How should you treat the negative edges.

Hope this helped! And I'm happy to discuss further.
